recirq/seniority_zero/circuits_simplified/core.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2023 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generic circuit routines"""
+from typing import List, Optional, Type, Generator
+
+import cirq
+
+from recirq.seniority_zero.circuits_expt.core import starting_pair_loop_on_lattice
+from recirq.seniority_zero.circuits_simplified.gates import GSGate
+
+
+def brickwall_loop_layer(
+    qubits: List[cirq.GridQubit], layer_params: List[float], offset: int, gate: Type[cirq.Gate]
+) -> Generator:
+    """Generates a layer of <gate> acting on a loop of qubits.
+
+    We call gate(layer_params[i])(qubits[(2 * i + offset) % len(qubits)],
+                                  qubits[(2 * i + offset + 1) % len(qubits)])
+    Args:
+        qubits [List[cirq.GridQubit]]: loop of qubits to act on
+        layer_params [List[float]]: list of parameters to call.
+        offset [int]: how far to shift along qubits
+        gate [cirq.Gate]: gate to be applied in brick wall
+    """
+    pairs = [
+        (qubits[(2 * ii + offset) % len(qubits)], qubits[(2 * ii + 1 + offset) % len(qubits)])
+        for ii in range(len(qubits))
+    ]
+    for param, (q0, q1) in zip(layer_params, pairs):
+        yield gate(param).on(q0, q1)
+
+
+def brickwall_givens_swap_network(
+    qubits: List[cirq.GridQubit], params: List[float], depth: int, inverse: Optional[bool] = False
+) -> Generator:
+    """Generates a brickwall Givens swap ansatz with fixed parameters.
+
+    Args:
+        qubits [List of cirq.GridQubit]: loop of qubits to act on
```

Comment:
List of qubits or a loop of qubits?

Reply:
Probably 'loop' here is better as this is the connectivity we assume. I've made a comment about this.